src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -0,0 +1,526 @@
+    return _scene->mNumMeshes;
+}
+
+std::optional<MeshData3D> AssimpImporter::doMesh3D(const UnsignedInt id) {
+    const aiMesh* mesh = _scene->mMeshes[id];
+
+    /* Primitive */
+    MeshPrimitive primitive;
+    if(mesh->mPrimitiveTypes == aiPrimitiveType_POINT) {
+        primitive = MeshPrimitive::Points;
+    } else if(mesh->mPrimitiveTypes == aiPrimitiveType_LINE) {
+        primitive = MeshPrimitive::Lines;
+    } else if(mesh->mPrimitiveTypes == aiPrimitiveType_TRIANGLE) {
+        primitive = MeshPrimitive::Triangles;
+    } else {
+        Error() << "Trade::AssimpImporter::mesh3D(): Unsupported aiPrimitiveType.";
+        return std::nullopt;
+    }
+
+    /* Mesh data */
+    std::vector<std::vector<Vector3>> positions;
+    std::vector<std::vector<Vector2>> textureCoordinates;
+    std::vector<std::vector<Vector3>> normals;
+    std::vector<std::vector<Color4>> colors;
+
+    /* Import positions */
+    positions.emplace_back(); // FIXME(C++17) assign to pos directly
+    auto& pos = positions[0];
+    pos.reserve(mesh->mNumVertices);
+    pos.insert(pos.end(), reinterpret_cast<Vector3*>(mesh->mVertices),
+            reinterpret_cast<Vector3*>(mesh->mVertices) + mesh->mNumVertices);
+
+    normals.emplace_back(); // FIXME(C++17) assign to pos directly
+    auto& norms = normals[0];
+    norms.reserve(mesh->mNumVertices);
+    norms.insert(norms.end(), reinterpret_cast<Vector3*>(mesh->mNormals),
+            reinterpret_cast<Vector3*>(mesh->mNormals) + mesh->mNumVertices*sizeof(Vector3));
+
+    /** @todo only first uv layer (or "channel") supported) */
+    textureCoordinates.reserve(mesh->GetNumUVChannels());
+    for (size_t layer = 0; layer < mesh->GetNumUVChannels(); ++layer) {
+        textureCoordinates.emplace_back();
+        auto& texCoords = textureCoordinates[layer];
+        texCoords.reserve(mesh->mNumVertices);
+        for(size_t i = 0; i < mesh->mNumVertices; ++i) {
+            /** GCC 4.7 has a problem with .x/.y here */
+            texCoords.emplace_back(mesh->mTextureCoords[layer][i][0], mesh->mTextureCoords[layer][i][1]);
+        }
+    }
+
+    colors.reserve(mesh->GetNumColorChannels());
+    for (size_t layer = 0; layer < mesh->GetNumColorChannels(); ++layer) {
+        colors.emplace_back();
+        auto& cols = colors[layer];
+        cols.reserve(mesh->mNumVertices);
+        cols.insert(cols.end(), reinterpret_cast<Color4*>(mesh->mColors[layer]),
+                reinterpret_cast<Color4*>(mesh->mColors[layer]) + mesh->mNumVertices*sizeof(Color4));
+    }
+
+    /* Import indices */
+    std::vector<UnsignedInt> indices;
+    indices.reserve(mesh->mNumFaces*3);
+
+    for(size_t faceIndex = 0; faceIndex < mesh->mNumFaces; ++faceIndex) {
+        const aiFace& face = mesh->mFaces[faceIndex];
+
+        CORRADE_ASSERT(face.mNumIndices == 3, "Expected number of indices per face to always be 3, since using triangulation post processing of assimp", {});
```

Comment:
Please prefix the assert message with the function name the same way as the error messages.